src/debug/debug.cc

```diff
@@ -4248,18 +4252,25 @@ GetOrCreatePromiseDependencyGraphData(Isolate* isolate, Handle<Object> promise)
   return iter->second;
 }
 
+extern bool gRecordReplayEnableDependencyGraph;
+
 bool RecordReplayShouldCallOnPromiseHook() {
   // The promise hook is normally only used when replaying, but can assign
   // persistent IDs to objects so needs to be called while recording if we are
   // asserting on these.
-  return (recordreplay::IsReplaying() || gRecordReplayAssertTrackedObjects)
+  return gRecordReplayEnableDependencyGraph
+      && (recordreplay::IsReplaying() || gRecordReplayAssertTrackedObjects)
       && IsMainThread();
 }
 
 void RecordReplayOnPromiseHook(Isolate* isolate, PromiseHookType type,
                                Handle<JSPromise> promise, Handle<Object> parent) {
   CHECK(RecordReplayShouldCallOnPromiseHook());
 
+  if (!gRecordReplayEnableDependencyGraph) {
+    return;
+  }
```

Comment:
These changes avoid running dependency graph related logic when it isn't enabled, allowing me to rule out some possibilities for the mismatches.